### PR TITLE
[Demangle] demangle builtin type transformations

### LINKED ISSUE #62127: clang generates symbol name which we can't demangle
With the attached source code [a.ii.gz](https://github.com/llvm/llvm-project/files/11231199/a.ii.gz), Clang generates a symbol which LLVM can't demangle:

```console
$ clang -cc1 -triple x86_64-unknown-linux-gnu -emit-obj -std=c++20 -w a.ii -S -emit-llvm -o - | grep declare.*__to_address_helper
declare noundef i32 @_ZNSt2Cr12__to_addressINS_16reverse_iteratorIPNS_4pairIjmEEEEvEEu7__decayIDTclsr19__to_address_helperIT_EE6__callclsr3stdE7declvalIRKS6_EEEEES8_(ptr noundef nonnull align 1 dereferenceable(1)) #1

$ llvm-cxxfilt _ZNSt2Cr12__to_addressINS_16reverse_iteratorIPNS_4pairIjmEEEEvEEu7__decayIDTclsr19__to_address_helperIT_EE6__callclsr3stdE7declvalIRKS6_EEEEES8_
_ZNSt2Cr12__to_addressINS_16reverse_iteratorIPNS_4pairIjmEEEEvEEu7__decayIDTclsr19__to_address_helperIT_EE6__callclsr3stdE7declvalIRKS6_EEEEES8_
```

(Originally from https://crbug.com/1078788#c22)

### PR DESCRIPTION
**Pull request by HerrCai0907**
Fixed: https://github.com/llvm/llvm-project/issues/62127 https://reviews.llvm.org/D116203 introduced several compiler builtin equivalents of the unary type traits. In some cases (e.g. template) those builtin will be dependent and need to be mangle.
This patch add the check for `u{builtin}I{type}E` to demangle it.

Reviewed By: rjmccall

Differential Revision: https://reviews.llvm.org/D148465

### CODE REVIEW

**Comment by shafik** [comment]
It looks like in: https://github.com/llvm/llvm-project/issues/62127 I identified that this was brought in via: https://github.com/llvm/llvm-project/commit/e9ef45635b77598fc9ce0cd38d7d3f8c9d88a49d

Do we need to add tests for the other builtins as well, just to verify that we don't have other issues? It is not obvious to me why this effects only `__decay`.

File: libcxxabi/test/test_demangle.pass.cpp [hunk 1]
```diff
@@ -30114,6 +30114,9 @@ const char* cases[][2] =
      " std::allocator<char>>::basic_string()"},
     {"_ZN1SB8ctor_tagC2Ev", "S[abi:ctor_tag]::S()"},
     {"_ZN1SB8ctor_tagD2Ev", "S[abi:ctor_tag]::~S()"},
+
+    // clang builtin type transform
+    {"_Z2f5IiEvu7__decayIT_E", "void f5<int>(__decay(int))"},
```
Comment (shafik) [inline libcxxabi/test/test_demangle.pass.cpp:30119]: Should we add tests to cover the other type transforms from: https://github.com/llvm/llvm-project/commit/e9ef45635b77598fc9ce0cd38d7d3f8c9d88a49d

**Comment by HerrCai0907** [comment]
@shafik Could it merge now?

**Comment by HerrCai0907** [comment]
merged due to already reviewed in phabricator

**Pull request merged**